site/src/pages/WorkspacePage/Workspace.tsx

```diff
@@ -257,6 +282,53 @@ export const Workspace: FC<WorkspaceProps> = ({
   );
 };
 
+const WorkspaceResourceData: FC<{ resource: TypesGen.WorkspaceResource }> = ({
+  resource,
+}) => {
+  const metadata = resource.metadata ?? [];
```

Comment:
Might be nice to spread `resource.metadata` into its own array so we aren't mutating what comes back from the server, i.e.
```
const metadata = [...(resource.metadata ?? [])]
```